File: AkitaModel/Results/VisitResults.cs
using System.Collections.Generic;

namespace Skyware.Lis.AkitaModel.Results;

/// <summary>
/// Represents a Visit view with structure for results reporting.
/// </summary>
public class VisitResults : VisitBase
{

    /// <summary>
    /// List of notes and announcements.
    /// </summary>
    public IList<Note> Notes { get; set; }

    /// <summary>
    /// List of departments with results.
    /// </summary>
    public IList<DepartmentResults> DepartmentResults { get; set; }

    /// <summary>
    /// List of samples used in examinations.
    /// </summary>
    public IList<Sample> Samples { get; set; }

    /// <summary>
    /// Footnotes list.
    /// </summary>
    public IList<Footnote> Footnotes { get; set; }

}

File: AkitaMotelTS/src/Results/VisitResults.ts
/**
 * Skyware.Lis.AkitaModel.Results
 */

import { VisitBase } from '../VisitBase';
import { Note } from './Note';
import { DepartmentResults } from './DepartmentResults';
import { Sample } from './Sample';
import { Footnote } from '../Footnote';

/**
 * Represents a Visit view with structure for results reporting.
 */
export class VisitResults extends VisitBase {
  /**
   * List of visit-level notes and announcements.
   */
  notes?: Note[];

  /**
   * List of departments with results.
   */
  departmentResults?: DepartmentResults[];

  /**
   * List of samples used in examinations.
   */
  samples?: Sample[];

  /**
   * Footnotes list.
   */
  footnotes?: Footnote[];
}
